contracts/LinniaRecords.sol

```diff
@@ -99,6 +102,33 @@ contract LinniaRecords is Ownable, Pausable, Destructible {
         return true;
     }
 
+    /// Add a record by user without any provider's signatures and get a reward.
+    ///
+    /// @param dataHash the hash of the data
+    /// @param metadata plaintext metadata for the record
+    /// @param dataUri the data uri path of the encrypted data
+    /// @param token the ERC20 token address for the rewarding token
+    function addRecordwithReward (
```

Comment:
Why a separate method? Nobody will choose the original method

Reply:
Not all use cases make sense with the reward.  For instance, a non-payable Smart contract may want to add records.